libcontainer/specconv/spec_linux.go

```diff
@@ -391,6 +396,21 @@ func convertSecToUSec(value dbus.Variant) (dbus.Variant, error) {
 	return dbus.MakeVariant(sec), nil
 }
 
+// getLinuxPersonalityFromStr converts the string domain received from spec to equivalent integer.
+// check libcontainer/configs/config_linux.go
+// check https://raw.githubusercontent.com/torvalds/linux/master/include/uapi/linux/personality.h
+// valid arguments
+// 	LINUX i.e PER_LINUX (since Linux 1.2.0).
+//   	LINUX32 i.e PER_LINUX32 (since Linux 2.2) LINUX32 will set the uname system call to show a 32 bit CPU type, such as i686.
+// 	everything unknown default to PER_LINUX.
+func getLinuxPersonalityFromStr(domain string) int {
+	// defaults to PER_LINUX
+	if domain == "LINUX32" {
+		return configs.PER_LINUX32
+	}
+	return configs.PER_LINUX
+}
```

Comment:
I don't think we should ignore strange settings -- this is just going to cause nightmares for backwards compatibility. Since `getLinuxPersonalityFromStr` will only be called if a user has *explicitly specified* a personality, we should definitely return an error if the personality name is meaningless.